tests/testthat/test-qenv_eval_code.R

```diff
@@ -172,3 +172,8 @@ testthat::test_that("comments passed alone to eval_code that contain @linksto ta
     "x"
   )
 })
+
+testthat::test_that("Code executed with integer shorthand (1L) is the same as original", {
+  q <- within(qenv(), a <- 1L)
+  testthat::expect_identical(get_code(q), "a <- 1L")
+})
```

Comment:
We don't need to cover if the syntax is correctly evaluated by `eval()`. We just need to make sure that the `code` is being treated asis. However, I've modified tests in eval_code to make sure your case is covered:

```r
  q <- qenv()
  q1 <- eval_code(q, quote(a <- 1L))
  q2 <- eval_code(q1, quote(b <- 1))
  testthat::expect_equal(q2@.xData, list2env(list(a = 1L, b = 1)))
```

Reply:
This works 😄. I get the distinction but I thought that was the easiest way to show that this worked and provide more assurance to a user. In addition, internal representation could change and would force to change the test too.

Reply:
@llrs-roche I got your points

> In addition, internal representation could change and would force to change the test too.

I'm happy you highlighted this problem. Since some time we focus on testing only public interface and internal representation should not be the reason to rewrite tests. This is why in most of the cases we use `get_code` instead of the `@code` or other equivalents.

> the easiest way to show that this worked and provide more assurance to a user

It is the easiest once you focus only on problem of `1L` being converted to numeric during deparsing in eval_code. This is well separated problem and with this test we ensure that code is deparsed corectly. 
Making sure that the result of the evaluation follows the code is a different problem and should be covered by another test - and it is in number of tests.

Reply:
I am not sure this was covered with other tests or it would have been caught before. But I take your point of making the test as minimal as possible and non overlapping. Thanks for the comments!